File: src/lib/translations.ts
// Frontend translations for EN/SK
export type Language = "en" | "sk";

export const translations = {
  en: {
    // Navigation
    nav: {
      dashboard: "Dashboard",
      subjects: "Subjects",
      enrolment: "Enrolment",
      schedule: "Schedule",
      grades: "Grades",
      assignments: "Assignments",
      dormitory: "Dormitory",
      thesis: "Thesis",
      payments: "Payments",
      notifications: "Notifications",
      profile: "Profile",
      activity: "Activity Log",
      settings: "Settings",
      documents: "Documents",
      logout: "Log out",
    },
    
    // Common
    common: {
      save: "Save",
      cancel: "Cancel",
      delete: "Delete",
      edit: "Edit",
      add: "Add",
      create: "Create",
      submit: "Submit",
      close: "Close",
      loading: "Loading...",
      saving: "Saving...",
      success: "Success",
      error: "Error",
      confirm: "Confirm",
      back: "Back",
      next: "Next",
      search: "Search",
      filter: "Filter",
      all: "All",
      none: "None",
      yes: "Yes",
      no: "No",
      download: "Download",
      upload: "Upload",
      view: "View",
      actions: "Actions",
    },
    
    // Auth
    auth: {
      login: "Login",
      logout: "Logout",
      register: "Register",
      email: "Email",
      password: "Password",
      confirmPassword: "Confirm Password",
      forgotPassword: "Forgot Password?",
      rememberMe: "Remember me",
      loginSuccess: "Login successful",
      loginError: "Invalid email or password",
      twoFactorRequired: "Two-factor authentication required",
      enterCode: "Enter 6-digit code",
      verifyCode: "Verify Code",
    },
    
    // Profile
    profile: {
      title: "Profile",
      subtitle: "Manage your account information",
      editProfile: "Edit Profile",
      changePassword: "Change Password",
      fullName: "Full Name",
      phone: "Phone",
      address: "Address",
      profilePicture: "Profile Picture",
      currentPassword: "Current Password",
      newPassword: "New Password",
      confirmNewPassword: "Confirm New Password",
      profileUpdated: "Profile updated successfully",
      passwordChanged: "Password changed successfully",
      role: "Role",
      accountStatus: "Account Status",
      active: "Active",
      inactive: "Inactive",
    },
    
    // Settings
    settings: {
      title: "Settings",
      subtitle: "Manage your preferences and account settings",
      appearance: "Appearance",
      theme: "Theme",
      darkMode: "Dark Mode",
      darkModeDesc: "Use dark theme for reduced eye strain",
      language: "Language",
      languageDesc: "Select your preferred language",
      timezone: "Timezone",
      timezoneDesc: "Set your local timezone",
      notifications: "Notifications",
      enableNotifications: "Enable Notifications",
      notificationsDesc: "Receive notifications about grades, schedules, and updates",
      security: "Security & Privacy",
      twoFactor: "Two-Factor Authentication",
      twoFactorDesc: "Add extra security to your account",
      enable: "Enable",
      disable: "Disable",
      enabled: "Enabled",
      disabled: "Disabled",
      saveChanges: "Save Changes",
      settingsSaved: "Settings saved successfully",
    },
    
    // Payments
    payments: {
      title: "Payments",
      subtitle: "Manage your tuition, dormitory, and other university payments",
      accountBalance: "Account Balance",
      amountDue: "Amount Due",
      allPaid: "All Paid",
      upcomingPayments: "Upcoming Payments",
      paymentHistory: "Payment History",
      paymentMethods: "Payment Methods",
      payNow: "Pay Now",
      createPayment: "Create Payment",
      invoiceNumber: "Invoice Number",
      dueDate: "Due Date",
      paidDate: "Paid Date",
      amount: "Amount",
      status: "Status",
      type: "Type",
      description: "Description",
      selectUser: "Select User",
      selectType: "Select Type",
      paid: "Paid",
      pending: "Pending",
      overdue: "Overdue",
      cancelled: "Cancelled",
      tuition: "Tuition Fee",
      dormitoryFee: "Dormitory Fee",
      adminFee: "Administrative Fee",
      bankTransfer: "Bank Transfer",
      creditCard: "Credit/Debit Card",
      cash: "Cash",
      paymentSuccess: "Payment successful",
      paymentCancelled: "Payment cancelled",
      downloadInvoice: "Download Invoice",
    },
    
    // Documents
    documents: {
      title: "Documents",
      subtitle: "Download official documents and upload submissions",
      officialDocuments: "Official Documents",
      myUploads: "My Uploads",
      uploadDocument: "Upload Document",
      enrollmentProof: "Proof of Enrollment",
      gradeTranscript: "Grade Transcript",
      downloadDesc: "Download official university documents",
      uploadDesc: "Upload documents for assignments or applications",
      selectFile: "Select File",
      documentType: "Document Type",
      uploadSuccess: "Document uploaded successfully",
      deleteSuccess: "Document deleted successfully",
      allowedTypes: "Allowed types: PDF, DOC, DOCX, JPEG, PNG",
      maxSize: "Maximum file size: 10MB",
    },
    
    // 2FA
    twoFactor: {
      setup: "Set up Two-Factor Authentication",
      scanQR: "Scan this QR code with your authenticator app",
      orEnterManually: "Or enter this code manually:",
      enterVerificationCode: "Enter verification code",
      verify: "Verify",
      disable: "Disable 2FA",
      disableConfirm: "Enter your current 2FA code to disable",
      setupSuccess: "Two-factor authentication enabled successfully",
      disableSuccess: "Two-factor authentication disabled",
      invalidCode: "Invalid verification code",
    },
    
    // Dashboard
    dashboard: {
      title: "Dashboard",
      welcome: "Welcome back",
      upcomingExams: "Upcoming Exams",
      recentGrades: "Recent Grades",
      todaySchedule: "Today's Schedule",
      notifications: "Notifications",
      quickActions: "Quick Actions",
    },
    
    // Grades
    grades: {
      title: "Grades",
      subtitle: "View your academic performance and grades",
      overallGPA: "Overall GPA",
      currentSemesterGPA: "Current Semester GPA",
      totalCredits: "Total Credits",
      gradeHistory: "Grade History",
      subject: "Subject",
      code: "Code",
      semester: "Semester",
      grade: "Grade",
      date: "Date",
      addGrade: "Add Grade",
      editGrade: "Edit Grade",
    },
  },
  
  sk: {
    // Navigation
    nav: {
      dashboard: "Prehľad",
      subjects: "Predmety",
      enrolment: "Zápis",
      schedule: "Rozvrh",
      grades: "Známky",
      assignments: "Zadania",
      dormitory: "Internát",
      thesis: "Záverečná práca",
      payments: "Platby",
      notifications: "Notifikácie",
      profile: "Profil",
      activity: "Aktivita",
      settings: "Nastavenia",
      documents: "Dokumenty",
      logout: "Odhlásiť sa",
    },
    
    // Common
    common: {
      save: "Uložiť",
      cancel: "Zrušiť",
      delete: "Vymazať",
      edit: "Upraviť",
      add: "Pridať",
      create: "Vytvoriť",
      submit: "Odoslať",
      close: "Zavrieť",
      loading: "Načítava sa...",
      saving: "Ukladá sa...",
      success: "Úspech",
      error: "Chyba",
      confirm: "Potvrdiť",
      back: "Späť",
      next: "Ďalej",
      search: "Hľadať",
      filter: "Filtrovať",
      all: "Všetko",
      none: "Žiadne",
      yes: "Áno",
      no: "Nie",
      download: "Stiahnuť",
      upload: "Nahrať",
      view: "Zobraziť",
      actions: "Akcie",
    },
    
    // Auth
    auth: {
      login: "Prihlásiť sa",
      logout: "Odhlásiť sa",
      register: "Registrovať sa",
      email: "Email",
      password: "Heslo",
      confirmPassword: "Potvrdiť heslo",
      forgotPassword: "Zabudnuté heslo?",
      rememberMe: "Zapamätať si ma",
      loginSuccess: "Prihlásenie úspešné",
      loginError: "Nesprávny email alebo heslo",
      twoFactorRequired: "Vyžaduje sa dvojfaktorová autentifikácia",
      enterCode: "Zadajte 6-miestny kód",
      verifyCode: "Overiť kód",
    },
    
    // Profile
    profile: {
      title: "Profil",
      subtitle: "Spravujte informácie o vašom účte",
      editProfile: "Upraviť profil",
      changePassword: "Zmeniť heslo",
      fullName: "Celé meno",
      phone: "Telefón",
      address: "Adresa",
      profilePicture: "Profilová fotka",
      currentPassword: "Aktuálne heslo",
      newPassword: "Nové heslo",
      confirmNewPassword: "Potvrdiť nové heslo",
      profileUpdated: "Profil bol úspešne aktualizovaný",
      passwordChanged: "Heslo bolo úspešne zmenené",
      role: "Rola",
      accountStatus: "Stav účtu",
      active: "Aktívny",
      inactive: "Neaktívny",
    },
    
    // Settings
    settings: {
      title: "Nastavenia",
      subtitle: "Spravujte svoje preferencie a nastavenia účtu",
      appearance: "Vzhľad",
      theme: "Téma",
      darkMode: "Tmavý režim",
      darkModeDesc: "Použite tmavú tému pre menšie namáhanie očí",
      language: "Jazyk",
      languageDesc: "Vyberte preferovaný jazyk",
      timezone: "Časové pásmo",
      timezoneDesc: "Nastavte vaše lokálne časové pásmo",
      notifications: "Notifikácie",
      enableNotifications: "Povoliť notifikácie",
      notificationsDesc: "Dostávajte notifikácie o známkach, rozvrhu a aktualizáciách",
      security: "Bezpečnosť a súkromie",
      twoFactor: "Dvojfaktorová autentifikácia",
      twoFactorDesc: "Pridajte extra zabezpečenie vášho účtu",
      enable: "Povoliť",
      disable: "Zakázať",
      enabled: "Povolené",
      disabled: "Zakázané",
      saveChanges: "Uložiť zmeny",
      settingsSaved: "Nastavenia boli úspešne uložené",
    },
    
    // Payments
    payments: {
      title: "Platby",
      subtitle: "Spravujte školné, ubytovanie a ďalšie univerzitné platby",
      accountBalance: "Zostatok účtu",
      amountDue: "Suma na úhradu",
      allPaid: "Všetko uhradené",
      upcomingPayments: "Nadchádzajúce platby",
      paymentHistory: "História platieb",
      paymentMethods: "Platobné metódy",
      payNow: "Zaplatiť teraz",
      createPayment: "Vytvoriť platbu",
      invoiceNumber: "Číslo faktúry",
      dueDate: "Dátum splatnosti",
      paidDate: "Dátum úhrady",
      amount: "Suma",
      status: "Stav",
      type: "Typ",
      description: "Popis",
      selectUser: "Vybrať používateľa",
      selectType: "Vybrať typ",
      paid: "Uhradené",
      pending: "Čakajúce",
      overdue: "Po splatnosti",
      cancelled: "Zrušené",
      tuition: "Školné",
      dormitoryFee: "Poplatok za internát",
      adminFee: "Administratívny poplatok",
      bankTransfer: "Bankový prevod",
      creditCard: "Kreditná/Debetná karta",
      cash: "Hotovosť",
      paymentSuccess: "Platba úspešná",
      paymentCancelled: "Platba zrušená",
      downloadInvoice: "Stiahnuť faktúru",
    },
    
    // Documents
    documents: {
      title: "Dokumenty",
      subtitle: "Stiahnite oficiálne dokumenty a nahrajte odovzdania",
      officialDocuments: "Oficiálne dokumenty",
      myUploads: "Moje nahrané súbory",
      uploadDocument: "Nahrať dokument",
      enrollmentProof: "Potvrdenie o štúdiu",
      gradeTranscript: "Výpis známok",
      downloadDesc: "Stiahnite oficiálne univerzitné dokumenty",
      uploadDesc: "Nahrajte dokumenty pre zadania alebo žiadosti",
      selectFile: "Vybrať súbor",
      documentType: "Typ dokumentu",
      uploadSuccess: "Dokument bol úspešne nahraný",
      deleteSuccess: "Dokument bol úspešne vymazaný",
      allowedTypes: "Povolené typy: PDF, DOC, DOCX, JPEG, PNG",
      maxSize: "Maximálna veľkosť súboru: 10MB",
    },
    
    // 2FA
    twoFactor: {
      setup: "Nastaviť dvojfaktorovú autentifikáciu",
      scanQR: "Naskenujte tento QR kód pomocou autentifikačnej aplikácie",
      orEnterManually: "Alebo zadajte tento kód manuálne:",
      enterVerificationCode: "Zadajte overovací kód",
      verify: "Overiť",
      disable: "Zakázať 2FA",
      disableConfirm: "Zadajte váš aktuálny 2FA kód pre zakázanie",
      setupSuccess: "Dvojfaktorová autentifikácia bola úspešne povolená",
      disableSuccess: "Dvojfaktorová autentifikácia bola zakázaná",
      invalidCode: "Neplatný overovací kód",
    },
    
    // Dashboard
    dashboard: {
      title: "Prehľad",
      welcome: "Vitajte späť",
      upcomingExams: "Nadchádzajúce skúšky",
      recentGrades: "Posledné známky",
      todaySchedule: "Dnešný rozvrh",
      notifications: "Notifikácie",
      quickActions: "Rýchle akcie",
    },
    
    // Grades
    grades: {
      title: "Známky",
      subtitle: "Zobrazte si váš akademický výkon a známky",
      overallGPA: "Celkový priemer",
      currentSemesterGPA: "Priemer aktuálneho semestra",
      totalCredits: "Celkové kredity",
      gradeHistory: "História známok",
      subject: "Predmet",
      code: "Kód",
      semester: "Semester",
      grade: "Známka",
      date: "Dátum",
      addGrade: "Pridať známku",
      editGrade: "Upraviť známku",
    },
  },
};

export function useTranslation(lang: Language) {
  return translations[lang] || translations.en;
}

export function t(lang: Language, path: string): string {
  const keys = path.split(".");
  let result: any = translations[lang] || translations.en;
  
  for (const key of keys) {
    result = result?.[key];
    if (result === undefined) {
      // Fallback to English
      result = translations.en;
      for (const k of keys) {
        result = result?.[k];
        if (result === undefined) return path;
      }
      break;
    }
  }
  
  return typeof result === "string" ? result : path;
}

File: backend/translations.py
"""
Translation helper for Slovak/English messages
"""

TRANSLATIONS = {
    "en": {
        # Auth
        "unauthorized": "Unauthorized access",
        "invalid_credentials": "Invalid email or password",
        "inactive_user": "Inactive user account",
        "email_registered": "Email already registered",
        "password_changed": "Password changed successfully",
        "invalid_password": "Invalid password",
        "passwords_not_match": "Passwords do not match",
        "password_too_short": "Password must be at least 6 characters",
        
        # 2FA
        "2fa_enabled": "Two-factor authentication enabled",
        "2fa_disabled": "Two-factor authentication disabled",
        "2fa_already_enabled": "2FA is already enabled",
        "2fa_not_enabled": "2FA is not enabled",
        "2fa_invalid_code": "Invalid verification code",
        "2fa_setup_required": "2FA setup not initiated",
        
        # Payments
        "payment_created": "Payment created successfully",
        "payment_successful": "Payment successful",
        "payment_cancelled": "Payment cancelled",
        "payment_not_found": "Payment not found",
        "payment_already_paid": "Payment already paid",
        "cannot_cancel_paid": "Cannot cancel a paid payment",
        
        # Documents
        "document_uploaded": "Document uploaded successfully",
        "document_deleted": "Document deleted successfully",
        "document_not_found": "Document not found",
        "file_not_found": "File not found on server",
        "file_type_not_allowed": "File type not allowed",
        
        # General
        "not_found": "Resource not found",
        "not_authorized": "Not authorized",
        "success": "Operation successful",
        "error": "An error occurred",
        "saved": "Changes saved",
        "deleted": "Deleted successfully",
        "updated": "Updated successfully",
        "created": "Created successfully",
        
        # User
        "user_not_found": "User not found",
        "profile_updated": "Profile updated successfully",
        "settings_updated": "Settings updated successfully",
        
        # Subjects
        "subject_not_found": "Subject not found",
        "already_enrolled": "Already enrolled in this subject",
        "enrollment_confirmed": "Enrollment confirmed",
        
        # Grades
        "grade_added": "Grade added successfully",
        "grade_updated": "Grade updated successfully",
        "grade_deleted": "Grade deleted successfully",
        
        # Dormitory
        "dormitory_not_found": "Dormitory not found",
        "application_submitted": "Application submitted successfully",
        "application_approved": "Application approved",
        "application_rejected": "Application rejected",
        
        # Thesis
        "thesis_not_found": "Thesis not found",
        "thesis_created": "Thesis created successfully",
        
        # Assignments
        "assignment_not_found": "Assignment not found",
        "submission_received": "Submission received successfully",
        "already_submitted": "You have already submitted this assignment",
    },
    
    "sk": {
        # Auth
        "unauthorized": "Neautorizovaný prístup",
        "invalid_credentials": "Nesprávny email alebo heslo",
        "inactive_user": "Neaktívny používateľský účet",
        "email_registered": "Email je už registrovaný",
        "password_changed": "Heslo bolo úspešne zmenené",
        "invalid_password": "Nesprávne heslo",
        "passwords_not_match": "Heslá sa nezhodujú",
        "password_too_short": "Heslo musí mať aspoň 6 znakov",
        
        # 2FA
        "2fa_enabled": "Dvojfaktorová autentifikácia bola povolená",
        "2fa_disabled": "Dvojfaktorová autentifikácia bola zakázaná",
        "2fa_already_enabled": "2FA je už povolená",
        "2fa_not_enabled": "2FA nie je povolená",
        "2fa_invalid_code": "Neplatný overovací kód",
        "2fa_setup_required": "Nastavenie 2FA nebolo iniciované",
        
        # Payments
        "payment_created": "Platba bola úspešne vytvorená",
        "payment_successful": "Platba úspešná",
        "payment_cancelled": "Platba bola zrušená",
        "payment_not_found": "Platba nebola nájdená",
        "payment_already_paid": "Platba už bola uhradená",
        "cannot_cancel_paid": "Nie je možné zrušiť uhradenú platbu",
        
        # Documents
        "document_uploaded": "Dokument bol úspešne nahraný",
        "document_deleted": "Dokument bol úspešne vymazaný",
        "document_not_found": "Dokument nebol nájdený",
        "file_not_found": "Súbor nebol nájdený na serveri",
        "file_type_not_allowed": "Typ súboru nie je povolený",
        
        # General
        "not_found": "Zdroj nebol nájdený",
        "not_authorized": "Neautorizovaný",
        "success": "Operácia úspešná",
        "error": "Vyskytla sa chyba",
        "saved": "Zmeny boli uložené",
        "deleted": "Úspešne vymazané",
        "updated": "Úspešne aktualizované",
        "created": "Úspešne vytvorené",
        
        # User
        "user_not_found": "Používateľ nebol nájdený",
        "profile_updated": "Profil bol úspešne aktualizovaný",
        "settings_updated": "Nastavenia boli úspešne aktualizované",
        
        # Subjects
        "subject_not_found": "Predmet nebol nájdený",
        "already_enrolled": "Už ste zapísaný v tomto predmete",
        "enrollment_confirmed": "Zápis potvrdený",
        
        # Grades
        "grade_added": "Známka bola úspešne pridaná",
        "grade_updated": "Známka bola úspešne aktualizovaná",
        "grade_deleted": "Známka bola úspešne vymazaná",
        
        # Dormitory
        "dormitory_not_found": "Internát nebol nájdený",
        "application_submitted": "Žiadosť bola úspešne podaná",
        "application_approved": "Žiadosť bola schválená",
        "application_rejected": "Žiadosť bola zamietnutá",
        
        # Thesis
        "thesis_not_found": "Záverečná práca nebola nájdená",
        "thesis_created": "Záverečná práca bola úspešne vytvorená",
        
        # Assignments
        "assignment_not_found": "Zadanie nebolo nájdené",
        "submission_received": "Odovzdanie bolo úspešne prijaté",
        "already_submitted": "Toto zadanie ste už odovzdali",
    }
}


def translate(key: str, lang: str = "en") -> str:
    """Get translated message for a key"""
    if lang not in TRANSLATIONS:
        lang = "en"
    return TRANSLATIONS.get(lang, {}).get(key, TRANSLATIONS["en"].get(key, key))


def get_all_translations(lang: str = "en") -> dict:
    """Get all translations for a language"""
    if lang not in TRANSLATIONS:
        lang = "en"
    return TRANSLATIONS.get(lang, TRANSLATIONS["en"])

